newIDE/app/src/ProjectManager/ExtensionsPropertiesDialog.js

```diff
@@ -0,0 +1,110 @@
+// @flow
+import { Trans } from '@lingui/macro';
+
+import * as React from 'react';
+import FlatButton from '../UI/FlatButton';
+import Dialog from '../UI/Dialog';
+import Text from '../UI/Text';
+import { ColumnStackLayout } from '../UI/Layout';
+import EmptyMessage from '../UI/EmptyMessage';
+import PropertiesEditor from '../PropertiesEditor';
+import propertiesMapToSchema from '../PropertiesEditor/PropertiesMapToSchema';
+import { List } from '@material-ui/core';
+
+type Props = {|
+  project: gdProject,
+  open: boolean,
+  onApply: Function,
+|};
+
+class ExtensionsPropertiesDialog extends React.Component<Props> {
+  _onApply = () => {
+    this.props.onApply();
+  };
+
+  _mapExtensionValueToPropertyDescriptors(
+    properties: gdMapStringPropertyDescriptor,
+    extensionName: string,
+    project: gdProject
+  ) {
+    const keys = properties.keys().toJSArray();
+    for (let key of keys) {
+      properties
+        .get(key)
+        .setValue(
+          project.getExtensionPropertiesManager().getValue(extensionName, key)
+        );
+    }
+    return properties;
+  }
+
+  render() {
+    const { project } = this.props;
+    const allExtensions = project
+      .getCurrentPlatform()
+      .getAllPlatformExtensions();
+    const properties = [];
+    for (let i = 0; i < allExtensions.size(); i++) {
+      const extension = allExtensions.at(i);
+      const propertiesSchema = propertiesMapToSchema(
```

Comment:
I think I did something wrong. When mounting the component I get an NaN error in console, and I cannot toggle booleans, it puts a warning about NaN in the console instead. Do you know where I have gone wrong?